_posts/2020-02-10-create-a-company-example.markdown

```diff
@@ -0,0 +1,114 @@
+---
+permalink: /v1/examples/creating-a-company
+title: Creating Companies
+layout: sidebar
+---
+
+# Creating a Company
+
+## Overview
+
+Gusto’s company provisioning API provides a way for partners to offer seamless Gusto account creation to their users.
+
+A successful request will create a new user and company in Gusto, will make the user the primary payroll administrator for the new company, and will send a welcome email to the new user.
+
+A successful response will return an account claim URL that the partner should redirect the user to so that they may finish setting up their account.
+
+## Authentication
+
+Due to the nature of this endpoint, Gusto will provide partners with an API token and will permit partners to use API Token Authentication instead of OAuth to provision Gusto accounts. The API token is included in the authorization HTTP header with the Token scheme, e.g.:
+
+```
+Content-Type: application/json
+Authorization: Token bbb286ff1a4fe6b84742b0d49b8d0d65bd0208d27d3d50333591df71
+```
+
+## Request Documentation
+
+| **Attribute**            | **Type**      | **Required** | **Description**
+| :----------              |:-------       |:-------      |:-------------
+| user                     | Object        | Y            | Information for the primary payroll administrator for the new company.
+| user:first_name          | String        | Y            | First name of the primary payroll admin.
+| user:last_name           | String        | Y            | Last name of the primary payroll admin.
+| user:email               | String        | Y            | Email for the primary payroll admin. This email will also be used for their login.
+| user:phone               | String        | N            | Phone number for the primary payroll admin.
+| company                  | Object        | Y            | Information for the new company.
+| company:name             | String        | Y            | Legal name of the company.
+| company:trade_name       | String        | N            | Name of the company.
+| company:ein              | String        | N            | FEIN of the company.
+| company:number_employees | Integer       | N            | The number of employees in the company.
+| company:states           | Array         | N            | The states in which the company operates.
+| company:addresses        | Array         | N            | The locations for the company. This includes mailing, work, and filing addresses.
```

Comment:
Good callouts. 

1. Yes, multiple addresses can be specified in the same request (i.e. work location and mailing/filing address). Right now there is not limit to how many addresses can be set, but I don't think good things would happen with an extremely large payload... 

	I'll make a ticket for that in the backlog unless you think it blocks the docs going live. 

2. If a single address is specified with `is_primary`, the address becomes a mailing, filing, and work location. If a single address is specified without `is_primary`, it would only become a work location. Do you think the docs need more clarification around this? 

	This makes me realize that we don't have validation to prevent two addresses from being marked as `is_primary`. If that were to happen, both would become work locations but the latter would become the mailing/filing address.

Reply:
1. I don't think we need to block the docs. A ticket is definitely a good idea! Even if we just block it on our side if it is > 5 or something like that.
2. I think you added the clarification already, and the clarification helps a lot! I wondered about validation around two `is_primary`. If there aren't use cases for two primary locations, we should file a ticket to put protections around that (or encourage the downstream team to do that in their model). If there are use cases, we should document what happens as you did here. Like above, I don't think this is a blocker for the docs going live.